jquery.nested_attributes.coffee

```diff
@@ -161,7 +161,7 @@ class NestedAttributes
       @bindDestroy($record) if @options.$clone or !@options.deepClone
 
       # Empty out the values of text inputs and selects
-      $record.find(':text, textarea, select').val('')
+      $record.find('input:not([type=submit]), textarea, select').val('')
```

Comment:
Sorry, I totally forgot about that. I have tested it and indeed it would cause checkboxes and radio buttons to clear values improperly. I have fixed this now in the updated commit. It is kinda long though. I can't find a shorter way to write it :(